ena_portal_api/ena_handler.py

```diff
@@ -55,26 +66,30 @@ def post_request(self, data):
             response = requests.post(self.url, data=data, **get_default_connection_headers())
         return response
 
-    def get_study(self, study_sec_acc):
+    # Supports ENA primary and secondary study accessions
+    def get_study(self, study_acc):
         data = get_default_params()
         data['result'] = 'study'
         data['fields'] = 'study_accession,secondary_study_accession,study_description,study_name,study_title,' \
                          'center_name,broker_name,last_updated,first_public'
-        data['query'] = 'secondary_study_accession=\"{}\"'.format(study_sec_acc)
+        if study_acc[0:2] in ('ERP', 'SRP', 'DRP'):
+            data['query'] = 'study_accession=\"{}\"'.format(study_acc)
+        else:
+            data['query'] = 'secondary_study_accession=\"{}\"'.format(study_acc)
         response = self.post_request(data)
         if str(response.status_code)[0] != '2':
```

Comment:
We might want to print out the status code itself and the response error class. That would help to investigate why we could not retrieve certain studies.